addons/mass_mailing/static/src/js/mass_mailing_html_field.js

```diff
@@ -194,6 +196,27 @@ export class MassMailingHtmlField extends HtmlField {
         }
 
         await this._resetIframe();
+
+        this.wysiwygEffectId = uuid();
+        const wysiwygEffectId = this.wysiwygEffectId;
+        effect(
+            batched(async (state) => {
```

Comment:
So... this seems a bit convoluted ^^

We're adding an "effect" (which I'm guessing is some kind of listener based on the value of the state variable? "batched" is also new) on the HTML field, that is based on a value from the wysiwyg, to check the status of the wysiwyg, to change values from the wysiwyg iframe.

I'm not sure why this can't be handled inside the "setSnippetsMenuFolded" method, which is in the wysiwyg, would seem a better place and much simpler.

But there is probably a good reason why we're doing this here.

Reply:
You're probably right that I can express this more simply and at a better place, I'll reevaluate it when I'm done with the other adjustments :)

Main idea here is to wait for the iframe to finish loading before applying the class.

indeed the effect callback is called when some subscribed property of state is written upon (and the subscription is done when that property is read on).

batched is there to only call the callback once in case multiple write are done on `state` synchronously (or else it would have been called as many time as there were writes)